src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -86,5 +140,68 @@ private Task OnFileWatcher(IFileItem file, object obj)
                 _poll = null;
             }
         }
+
+        private async Task<string> GetLastPollStatusAsync()
+        {
+            try
+            {
+                var apiHubBlob = GetBlobReference();
+
+                return await apiHubBlob.DownloadTextAsync();
+            }
+            catch (StorageException exception)
+            {
+                if (exception.RequestInformation == null ||
+                    exception.RequestInformation.HttpStatusCode != 404)
+                {
+                    var errorText = string.Format("Encountered a storage exception while getting the next poll status from the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, exception.Message);
+                    _trace.Error(errorText);
+                }
+                return null;
+            }
+            catch (Exception e)
+            {
+                var errorText = string.Format("Encountered an exception while getting the next poll status from the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, e.Message);
+                _trace.Error(errorText);
+                return null;
+            }
+        }
+
+        private async Task SetNextPollStatusAsync(string nextPoll)
+        {
+            try
+            {
+                var apiHubBlob = GetBlobReference();
+                await apiHubBlob.UploadTextAsync(nextPoll);
+            }
+            catch (Exception e)
+            {
+                var errorText = string.Format("Encountered an exception while setting the next poll status to the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, e.Message);
+                _trace.Error(errorText);
+            }
+        }
+
+        private CloudBlockBlob GetBlobReference()
+        {
+            // Path to the apiHub blob is:
+            // apihubs/{siteName}/{functionaName}/{triggerPath}/{connectionStringSetting}
+            // This needs to be kept in sync with Antares Scale Controller
+            string blobName = string.Format("{0}/{1}/{2}", this._functionName, this._folderSource.Path, this._connectionStringSetting);
+            return this._apiHubBlobDirectory.GetBlockBlobReference(blobName);
+        }
+
+        private static string GetSiteName(JobHostConfiguration config)
+        {
+            var siteName = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME");
```

Comment:
Here is the logic: https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script.WebHost/App_Start/AutofacBootstrap.cs#L24